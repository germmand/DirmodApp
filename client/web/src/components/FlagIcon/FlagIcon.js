import React from 'react';
import PropTypes from 'prop-types';

import BR_FLAG from '../../assets/flags/BR.svg';
import EU_FLAG from '../../assets/flags/EU.svg';
import US_FLAG from '../../assets/flags/US.svg';

const Flags = {
  "BRL": BR_FLAG,
  "EUR": EU_FLAG,
  "USD": US_FLAG,
};

const FlagIcon = ({ code, style, size }) => (
  <img
    alt={code}
    src={Flags[code]}
    style={{
      ...style,
      height: `${size}px`,
      width: `${size}px`,
      display: 'inline-block'
    }}
  />
);

FlagIcon.propTypes = {
  code: PropTypes.string.isRequired,
  size: PropTypes.number.isRequired,
  style: PropTypes.objectOf(PropTypes.string),
};

FlagIcon.defaultProps = {
  style: {},
};

export default FlagIcon;
