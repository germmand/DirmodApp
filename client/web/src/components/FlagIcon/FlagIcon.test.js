import React from 'react';
import renderer from 'react-test-renderer';

import FlagIcon from './FlagIcon';

describe('FlagIcon Component', () => {
  it('should match snapshot', () => {
    const props = {
      code: 'EUR',
      size: 25,
    };
    const component = renderer.create(
      <FlagIcon {...props} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});