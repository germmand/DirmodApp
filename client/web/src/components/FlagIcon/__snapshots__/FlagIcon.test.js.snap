// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlagIcon Component should match snapshot 1`] = `
<img
  alt="EUR"
  src="EU.svg"
  style={
    Object {
      "display": "inline-block",
      "height": "25px",
      "width": "25px",
    }
  }
/>
`;
